src/sidebar/zeplinComponent/data/CumulativeBarrelDetailsStore.ts

```diff
@@ -0,0 +1,52 @@
+import BarrelDetails from "../../../common/domain/zeplinComponent/model/BarrelDetails";
+import Store from "../../../common/domain/store/Store";
+import BarrelError from "../../../common/domain/zeplinComponent/model/BarrelError";
+import BarrelType from "../../../common/domain/barrel/BarrelType";
+import Result from "../../../common/domain/store/Result";
+import BarrelDetailsStoreProvider from "../../../common/domain/zeplinComponent/data/BarrelDetailsStoreProvider";
+import { convertBarrelResult } from "../../../common/domain/barrel/util/barrelUtil";
+
+export default class CumulativeBarrelDetailsStore implements Store<BarrelDetails[], BarrelError> {
+    public constructor(private barrelId: string, private barrelType: BarrelType) { }
+
+    public get = async (): Promise<Result<BarrelDetails[], BarrelError>> => {
+        const leafId = this.barrelId;
+        const leafType = this.barrelType;
+        let currentId: string | undefined = leafId;
+        let currentType = leafType;
+        const accumulatedResult: Result<BarrelDetails[], BarrelError> = {
+            data: [],
+            errors: []
+        };
+
+        while (currentId) {
+            const childId: string | undefined = currentId === leafId ? undefined : leafId;
+            const childType: BarrelType | undefined = currentId === leafId ? undefined : leafType;
```

Comment:
Types with ternary looks so confusing here. Can't we define type of `leafId` above where we define it? We already know undefined is undefined 😄

Reply:
I tried that now, the result is: 

> 'childType' implicitly has type 'any' because it does not have a type annotation and is referenced directly or indirectly in its own initializer. ts(7022)

We have to explicitly set it.